### PR TITLE
[HORRIBLY NONMODULAR] Pain (of the surgical variety)

### PR DESCRIPTION
**Pull request by FlamingLily**
# NOTE: https://github.com/Skyrat-SS13/Skyrat-tg/pull/5858 is currently waiting merge. I recommend holding off on merging until that PR is merged and I resolve the conflicts, BUT, if you want to, merge it and deal with the conflicts if you're some kinda masochist
## About The Pull Request
Adds unique pain flavour text to each surgical step. Short, mundane steps have 1 message, such as incisions. Long, critical steps have unique messages for pre-op, success and failure, such as brain surgery.

## Why It's Good For The Game
Fosters RP, Encourages medbay to actually dang use their tools.

## Changelog
:cl:
add: Surgical pain messages to every single surgery.
/:cl:

### CODE REVIEW

**Comment by FlamingLily** [comment]
The reason I did this non-modularly is due to how much of the surgery code it edits. I easily could modularise it, but I'd be taking so much of the surgery code out and into modular files that at that point we may as well port bobmed 😼 (i mean thats still on the table)

If maintainers want me to modularise, I'll do so

**Comment by thestubborn** [comment]
> 
> 
> The reason I did this non-modularly is due to how much of the surgery code it edits. I easily could modularise it, but I'd be taking so much of the surgery code out and into modular files that at that point we may as well port bobmed 😼 (i mean thats still on the table)
> 
> If maintainers want me to modularise, I'll do so

do it, port bobmed

**Comment by Cheshify** [comment]
Couldn't a lot of this go upstream?

**Comment by FlamingLily** [comment]
> 
> 
> Couldn't a lot of this go upstream

Mmmaybe, i'll ask

**Comment by NullDagaf** [comment]
> do it, port bobmed

"Wahh, too punishing, revert"

Also, asking the original creator would probably be wiser.

**Comment by NullDagaf** [comment]
> Couldn't a lot of this go upstream?

pain

**Comment by Cheshify** [comment]
I don't think /tg/ has an issue with flavour at all, especially if it's well written and makes medbay a little more intuitive. This is a good PR, but I think 38 files worth of nonmodularity is a little too much for something our upstream would probably accept.
I get it though, PRing upstream sucks sometimes.

**Comment by NullDagaf** [comment]
> I don't think /tg/ has an issue with flavour at all, especially if it's well written and makes medbay a little more intuitive. This is a good PR, but I think 38 files worth of nonmodularity is a little too much for something our upstream would probably accept.
> I get it though, PRing upstream sucks sometimes.

I'm mostly opposed to that, since it takes away any originality a server could have, making it instead "something common". It's not that special if we have a whole new job, if most servers have that too, y'know?

**Comment by Cheshify** [comment]
> I'm mostly opposed to that, since it takes away any originality a server could have, making it instead "something common". It's not that special if we have a whole new job, if most servers have that too, y'know?

Now I disagree with this, if you've got a positive change that can only really be done in a massively nonmodular way, it's better to share upstream and make both servers better. It's an open source game, and Skyrat has enough defining features that our playerbase won't rush to /tg/ because we shared a flavour PR.

File: code/modules/surgery/advanced/bioware/cortex_folding.dm [hunk 1]
```diff
@@ -27,11 +27,13 @@
 	display_results(user, target, "<span class='notice'>You start folding [target]'s outer cerebral cortex into a fractal pattern.</span>",
 		"<span class='notice'>[user] starts folding [target]'s outer cerebral cortex into a fractal pattern.</span>",
 		"<span class='notice'>[user] begins to perform surgery on [target]'s brain.</span>")
+	display_pain(target, "<span class='userdanger'>Your head throbs with gruesome pain, it's nearly too much to handle!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/cortex_folding.dm:30]: Move that to modular, copy the proc over, put that line in and call ..()
```suggestion
datum/surgery_step/fold_cortex/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
	display_pain(target, "<span class='userdanger'>Your head throbs with gruesome pain, it's nearly too much to handle!</span>")
	..()
```
...don't accept this as a valid change, I just can't write in a different way on phone.

File: code/modules/surgery/advanced/bioware/cortex_folding.dm [hunk 2]
```diff
@@ -27,11 +27,13 @@
...
 
 /datum/surgery_step/fold_cortex/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
 	display_results(user, target, "<span class='notice'>You fold [target]'s outer cerebral cortex into a fractal pattern!</span>",
 		"<span class='notice'>[user] folds [target]'s outer cerebral cortex into a fractal pattern!</span>",
 		"<span class='notice'>[user] completes the surgery on [target]'s brain.</span>")
+	display_pain(target, "<span class='userdanger'>Your brain feels stronger... more flexible!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/cortex_folding.dm:36]: Same here.

File: code/modules/surgery/advanced/bioware/cortex_imprint.dm [hunk 3]
```diff
@@ -27,11 +27,13 @@
 	display_results(user, target, "<span class='notice'>You start carving [target]'s outer cerebral cortex into a self-imprinting pattern.</span>",
 		"<span class='notice'>[user] starts carving [target]'s outer cerebral cortex into a self-imprinting pattern.</span>",
 		"<span class='notice'>[user] begins to perform surgery on [target]'s brain.</span>")
+	display_pain(target, "<span class='userdanger'>Your head throbs with gruesome pain, it's nearly too much to handle!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/cortex_imprint.dm:30]: Same here.

File: code/modules/surgery/advanced/bioware/cortex_imprint.dm [hunk 4]
```diff
@@ -27,11 +27,13 @@
...
 
 /datum/surgery_step/imprint_cortex/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
 	display_results(user, target, "<span class='notice'>You reshape [target]'s outer cerebral cortex into a self-imprinting pattern!</span>",
 		"<span class='notice'>[user] reshapes [target]'s outer cerebral cortex into a self-imprinting pattern!</span>",
 		"<span class='notice'>[user] completes the surgery on [target]'s brain.</span>")
+	display_pain(target, "<span class='userdanger'>Your brain feels stronger... more resilient!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/cortex_imprint.dm:36]: Same here.

File: code/modules/surgery/advanced/bioware/ligament_hook.dm [hunk 5]
```diff
@@ -21,11 +21,13 @@
 	display_results(user, target, "<span class='notice'>You start reshaping [target]'s ligaments into a hook-like shape.</span>",
 		"<span class='notice'>[user] starts reshaping [target]'s ligaments into a hook-like shape.</span>",
 		"<span class='notice'>[user] starts manipulating [target]'s ligaments.</span>")
+	display_pain(target, "<span class='userdanger'>Your limbs burn with severe pain!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/ligament_hook.dm:24]: Same here.

File: code/modules/surgery/advanced/bioware/ligament_hook.dm [hunk 6]
```diff
@@ -21,11 +21,13 @@
...
 
 /datum/surgery_step/reshape_ligaments/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, default_display_results = FALSE)
 	display_results(user, target, "<span class='notice'>You reshape [target]'s ligaments into a connective hook!</span>",
 		"<span class='notice'>[user] reshapes [target]'s ligaments into a connective hook!</span>",
 		"<span class='notice'>[user] finishes manipulating [target]'s ligaments.</span>")
+	display_pain(target, "<span class='userdanger'>Your limbs feel... strangely loose.</span>") //SKYRAT EDIT ADD - SURGERY PAIN
 	new /datum/bioware/hooked_ligaments(target)
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/ligament_hook.dm:31]: Same here.

File: code/modules/surgery/advanced/bioware/ligament_reinforcement.dm [hunk 7]
```diff
@@ -21,11 +21,13 @@
 	display_results(user, target, "<span class='notice'>You start reinforcing [target]'s ligaments.</span>",
 		"<span class='notice'>[user] starts reinforce [target]'s ligaments.</span>",
 		"<span class='notice'>[user] starts manipulating [target]'s ligaments.</span>")
+	display_pain(target, "<span class='userdanger'>Your limbs ache with intense pain!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/ligament_reinforcement.dm:24]: Same here.

File: code/modules/surgery/advanced/bioware/muscled_veins.dm [hunk 8]
```diff
@@ -20,11 +20,13 @@
 	display_results(user, target, "<span class='notice'>You start wrapping muscles around [target]'s circulatory system.</span>",
 		"<span class='notice'>[user] starts wrapping muscles around [target]'s circulatory system.</span>",
 		"<span class='notice'>[user] starts manipulating [target]'s circulatory system.</span>")
+	display_pain(target, "<span class='userdanger'>Your entire body burns in agony!</span>") //SKYRAT EDIT ADD - SURGERY PAIN
```
Comment (NullDagaf) [inline code/modules/surgery/advanced/bioware/muscled_veins.dm:23]: Same here.

**Comment by NullDagaf** [comment]
Okay, there's a lot more places that this applies to, and I can't be bothered to track all of them at 3 AM.

**Comment by NullDagaf** [comment]
> Now I disagree with this, if you've got a positive change that can only really be done in a massively nonmodular way, it's better to share upstream and make both servers better. It's an open source game, and Skyrat has enough defining features that our playerbase won't rush to /tg/ because we shared a flavour PR.

Well yes, except as I've shown, this can be modularised a lot more.
Also non-modularity is only an issue if you're unwilling to maintain the code you're editing. _And you wouldn't abandon your own code, now would you?_

**Comment by NullDagaf** [comment]
> do it, port bobmed

(Also p sure bobmed is the pinnacle of non-modularity 🤣 )

**Comment by FlamingLily** [comment]
> 
> 
> Move that to modular, copy the proc over, put that line in and call ..()

i may be stupid

**Comment by FlamingLily** [comment]
hahah time for conflict pain https://github.com/Skyrat-SS13/Skyrat-tg/pull/5858

**Comment by NullDagaf** [comment]
> i may be stupid

Inexperienced at most. But I wouldn't call this stupid.

**Comment by Gandalf2k15** [comment]
Honestly, let's not do it like this. I'd rather we redid the medical system fully. We should start a discussion with TG about it.

**Pull request closed without merging**